test/jasmine/tests/image_test.js

```diff
@@ -669,5 +667,30 @@ describe('image hover:', function() {
                 .then(done);
             });
         });
+
+        [
+            [[0, 512], [0, 512]],
+            [[0, 512], [512, 0]],  // the default image layout
+            [[512, 0], [0, 512]],
+            [[512, 0], [512, 0]]
+        ].forEach(function(test) {
+            it('should show correct hover info regardless of axis directions ' + test, function(done) {
+                var mockCopy = Lib.extendDeep({}, mock);
+                mockCopy.layout.xaxis.range = test[0];
+                mockCopy.layout.yaxis.range = test[1];
+                mockCopy.data[0].colormodel = 'rgba';
+                mockCopy.data[0].hovertemplate = '%{z}<extra></extra>';
+                Plotly.newPlot(gd, mockCopy)
+                .then(function() {_hover(205, 125);})
+                .then(function() {
+                    assertHoverLabelContent({
+                        nums: '[202, 148, 125, 255]',
+                        name: ''
+                    }, 'variable `z` should be correct!');
+                })
+                .catch(failTest)
+                .then(done);
+            });
+        });
```

Comment:
Does this look OK? A the moment only the 2nd iteration passes, suggesting that the hover does not take the axis orientation into account, even though it *does* work correctly when interacting with a figure in the browser. Could it be that the `_hover()` here does not follow the exact same code path?